DAL/BaseDao.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.Data.SqlClient;
+using System.Configuration;
+using System.Data;
+
+namespace DAL
+{
+    internal abstract class BaseDao
+    {
+        const string ConnectionStringName = "ChapeauDatabase";
+        const string DatabaseErrorMessage = "Database operation failed.";
+        protected const int Zero = 0;
+
+        private SqlConnection OpenConnection()
+        {
+            SqlConnection connection = new SqlConnection(ConfigurationManager.ConnectionStrings[ConnectionStringName].ConnectionString);
+            connection.Open();
+            return connection;
+        }
+
+        protected List<T> ReadTable<T>(DataTable dataTable, Func<DataRow, T> readRow)
```

Comment:
Maybe make BaseDao<T> instead of specifying data type in the method like ReadTable<T>. The code will look cleaner specially if we will have more generic methods